pkg/sql/show_create_external_connection.go

```diff
@@ -58,9 +62,34 @@ func loadExternalConnections(
 	}
 
 	for _, row := range rows {
-		connectionName := tree.MustBeDString(row[0])
+		name := string(tree.MustBeDString(row[0]))
+
+		if checkUsagePrivilege {
+			// Check that the user has USAGE privileges on the External Connection object.
+			ecPrivilege := &syntheticprivilege.ExternalConnectionPrivilege{
+				ConnectionName: name,
+			}
+			hasPriv, err := params.p.HasPrivilege(params.ctx, ecPrivilege,
+				privilege.USAGE, params.p.User())
+			if err != nil {
+				return nil, err
+			}
+			if !hasPriv {
+				continue
+			}
```

Comment:
🤔 I wonder if we should return an error in the case where ConnectionLabel != nil.

Reply:
Agree!